### DISCUSSION TITLE
LCD Display from Sunton with Speaker

### DISCUSSION THREAD

**Discussion by wilwizard1979** [HASwitchPlate/openHASP#469] [Hardware]
I have an LCD model Sunton E3248S035C.
Is there any way to use the Speaker output?
Since I’m using OpenHASP as a room HMI control, it would be great to use the built-in Speaker output? OpenHASP could for example play a Radio through Home Assistant. 
The last might be complicated since HA would need to stream the audio to the device, I guess. But even a simpler solution to play .MP3 files fro. The SD Card could be used to for example play a House alarm in all rooms.

**Comment by fvanroie** [comment]
> Is there any way to use the Speaker output?

There are no audio drivers implemented in the standard firmware. You would need to implement the driver and compile the firmware locally.

> The SD Card could be used to for example play a House alarm in all rooms.

The same goes for the SD filesystem...

**Reply by wilwizard1979** [reply]
Thanks @fvanroie. I would like to try but would need some initial support to kick-off.
I started my professional career writing code sometimes in Assembly, Pascal, Delphi and VB to connect equipment in the automation industry. Sometimes critical communication to operate a metallurgy rolling mill for example. Later I continued with Process Automation in the mining industry but never experienced building a IT driver. As Hobby I´ve also played around with Arduino development board.
If you can give me some hints where to start, I´m motivated enough to try this.

**Comment by broyuken** [comment]
Did you ever get anywhere with this? I have a wt32-sc01 plus and wanted to look at playing audio from home assistant on openhasp.

**Comment by elik745i** [comment]
Interested too!

**Comment by xNUTx** [comment]
likewise :)

**Comment by darkman7hr** [comment]
in for playing mp3 from sd card. Sunton.

**Comment by darkman7hr** [comment]
did anyone made custom? :)

**Comment by hb020** [comment]
The next coming weeks I probably have some time to work out audio output with the built-in amp in the SC01-plus.
If possible via ESP32-audioI2s, but if impossible (likely), via a slimmed down version that can play files from local storage. The latter will likely mean: limited sample rates, few file formats.

@fvanroie Do you see any limitations around the use of the [ESP32 native I2S functions](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/peripherals/i2s.html#inter-ic-sound-i2s)? The native SDK uses event callbacks, interrupts and DMA.

**Reply by fvanroie** [reply]
I haven't used any I2S peripherals, so I am not aware of any limitations.

**Comment by hb020** [comment]
That Sunton device uses DAC channel 2 (IO26) to directly play audio. This is totally different from my device, SC01-Plus, that uses I2S. But you might find inspiration here: #817
